src/snowflake/snowpark/table.py

```diff
@@ -270,27 +270,36 @@ def __init__(
         self,
         table_name: str,
         session: Optional["snowflake.snowpark.session.Session"] = None,
+        is_generated_temp_table: bool = False,
     ) -> None:
-        super().__init__(
-            session, session._analyzer.resolve(UnresolvedRelation(table_name))
-        )
-        self.is_cached: bool = self.is_cached  #: Whether the table is cached.
-        self.table_name: str = table_name  #: The table name
-
-        if self._session.sql_simplifier_enabled:
-            self._select_statement = session._analyzer.create_select_statement(
+        if session.sql_simplifier_enabled:
+            plan = session._analyzer.create_select_statement(
                 from_=session._analyzer.create_selectable_entity(
-                    table_name, analyzer=session._analyzer
+                    table_name,
+                    analyzer=session._analyzer,
```

Comment:
I see why you are changing the selectable now, can we update the SelectEntity object to point to the SnowflakeTable object instead of just a name ? which seems a cleaner way to me

Reply:
good point, done